diff --git a/CMakeLists.txt b/CMakeLists.txt
index 523faa7..2523b22 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -336,8 +336,6 @@ check_cxx_source_compiles("
   int main() {}" MP_VARIADIC_TEMPLATES)
 
 if (MP_VARIADIC_TEMPLATES)
-  add_executable(nl-example src/nl-example.cc)
-  target_link_libraries(nl-example mp)
 endif ()
 
 add_subdirectory(doc)
@@ -384,13 +384,8 @@ endforeach ()
 if (MP_VARIADIC_TEMPLATES AND HAVE_ATOMIC)
   option(BENCHMARK_ENABLE_TESTING
          "Enable testing of the benchmark library." OFF)
-  add_subdirectory(thirdparty/benchmark)
-  enable_cxx11(benchmark)
 endif ()
 
-enable_testing()
-add_subdirectory(test)
-
 install(DIRECTORY include/mp DESTINATION include)
 install(TARGETS mp DESTINATION lib RUNTIME DESTINATION bin)
 install(FILES LICENSE.rst DESTINATION share/mp)
diff --git a/include/mp/format.h b/include/mp/format.h
index c5d09b5..4f5f20e 100644
--- a/include/mp/format.h
+++ b/include/mp/format.h
@@ -1747,21 +1747,21 @@ backport of fmtlib/fmt abbefd7; see fmtlib/fmt#398
     typedef typename BasicWriter<Char>::CharPtr CharPtr;
     Char fill = internal::CharTraits<Char>::cast(spec_.fill());
     CharPtr out = CharPtr();
-    const unsigned CHAR_WIDTH = 1;
-    if (spec_.width_ > CHAR_WIDTH) {
+    const unsigned CHAR_SIZE = 1;
+    if (spec_.width_ > CHAR_SIZE) {
       out = writer_.grow_buffer(spec_.width_);
       if (spec_.align_ == ALIGN_RIGHT) {
-        std::uninitialized_fill_n(out, spec_.width_ - CHAR_WIDTH, fill);
-        out += spec_.width_ - CHAR_WIDTH;
+        std::uninitialized_fill_n(out, spec_.width_ - CHAR_SIZE, fill);
+        out += spec_.width_ - CHAR_SIZE;
       } else if (spec_.align_ == ALIGN_CENTER) {
         out = writer_.fill_padding(out, spec_.width_,
-                                   internal::check(CHAR_WIDTH), fill);
+                                   internal::check(CHAR_SIZE), fill);
       } else {
-        std::uninitialized_fill_n(out + CHAR_WIDTH,
-                                  spec_.width_ - CHAR_WIDTH, fill);
+        std::uninitialized_fill_n(out + CHAR_SIZE,
+                                  spec_.width_ - CHAR_SIZE, fill);
       }
     } else {
-      out = writer_.grow_buffer(CHAR_WIDTH);
+      out = writer_.grow_buffer(CHAR_SIZE);
     }
     *out = internal::CharTraits<Char>::cast(value);
   }
